packages/x-data-grid-premium/src/DataGridPremium/DataGridPremium.tsx

```diff
@@ -975,6 +975,17 @@ DataGridPremiumRaw.propTypes = {
     PropTypes.number,
     PropTypes.string,
   ]),
+  /**
+   * The following behavior happens for each of the possible values:
+   * 1. `none` - No row selection propagation.
+   * 2. `parents` - Selecting all children will auto-select the parent(s).
+   * 3. `children` - Selecting a parent will auto-select all its descendants.
+   * 4. `both` - Both `parents` and `children` behavior.
```

Comment:
I have some problem with the current name of the options, I don't think they provide a good readability if you don't know the feature well.

- Not a big fan of "both" because you can't understand what it mean without knowing the other options.

- Why calling it `children` if the actual behavior is `descendants`? Wouldn't `descendants` be more explicit? One would make the same argument for `parents` vs `ancestors` but I think this is less problematic.

____

A few alternatives:

- `up` / `down` and `bidirectional` => main weakness is that "up" could mean the siblings visually above instead of the element "up the tree".
- Change the format to be `rowSelectionPropagation?: { parents?: boolean, descendants?: boolean }`

Reply:
I agree, the current version isn't very discoverable/readable.
Thank you for the suggestions. I really like the second one. There could be some confusion though, for example:

```js
rowSelectionPropagation: undefined;
```

```js
rowSelectionPropagation: {}
```

```js
rowSelectionPropagation: { parents: false, descendants: false }
```

All these would essentially mean `rowSelectionPropagation: "none"`. Also, storing more object references might cause application memory usage to grow a bit.

Another option could be to use two distinct props of `boolean` type with clear descriptions.

```js
autoSelectParents: boolean;
autoSelectChildren: boolean;
```

Would that be simpler/less confusing? 🤔

Reply:
> Using object may also increase object allocations in the application.

It's a single object, there isn't one per row or one per cell so I don't think it's very problematic.

___

> All these would essentially mean rowSelectionPropagation: "none"

I agree that this is the main challenge.
You can also have the following:

```tsx
/**
 * @default { parents: false, descendants: false }
 */
rowSelectionPropagation?: { parents: boolean, descendants: boolean }
```

That way you only have one way of describing each state (plus `undefined` but like any optional prop their is a default value).

___

 > Another option could be to use two distinct props of boolean type. Adding clear descriptions that explain their usage would reduce some friction.

If all the combinations are viable (which seems to be the case here), then it's an interesting alternative.
It's the same logic as the object I proposed but with a slightly different DX. 
No big preference here.

Reply:
> It's a single object, there isn't one per row or one per cell so I don't think it's very problematic.

True 👍

To clarify, for me, the motivation behind mentioning the object allocations is its possible impact on memory (both RAM and bundle size).

For example, on analyzing [this](https://bundlephobia.com/package/@mui/x-data-grid@7.18.0) bundle, I was amazed at the memory consumed by a static object `DATA_GRID_PROPS_DEFAULT_VALUES` (gzipped 52.6kB) which would grow even more if we keep adding complex defaults (such as objects) to it. The same would be the case for RAM usage. For reference, a `boolean` would take a max 4-5 bytes in memory while the default object `{ parents: false, descendants: false }` would take ~60 bytes.

This might be a micro-optimization, but many such combined may make some considerable impact. It shouldn't be done at the cost of compromising the DX though.

> It's the same logic as the object I proposed but with a slightly different DX.
No big preference here.

Yes, it's more of a DX question. For me as a library user, having distinct props sounds like a better DX which would provide a separation of concerns and divide the feature into two interrelated sub-features/categories.

I'd love to hear more thoughts on it.

Reply:
> AG grid is also covering filtered descendants, which presents some interesting use cases. Is it a quick win to support it?

@joserodolfofreitas For us, it is the default behavior to only select the filtered descendants, we even go one step further to deselect non-filtered rows on filtering automatically to avoid bugs like #11726 

So doing `rowSelectionPropagation.descendants=true` would already mean filtered descendants.

I'd be interested to know what you think around this. For me, I'll prefer to have some actual use-case raised before supporting selecting all descendants (ag-grid option 'descendants')

Reply:
Sounds great! Applying only to filtered is a perfect start! 💯 In favor of keeping like this.